src/components/ModelActions/ModelActions.test.tsx

```diff
@@ -0,0 +1,148 @@
+import { screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import {
+  configFactory,
+  generalStateFactory,
+  authUserInfoFactory,
+} from "testing/factories/general";
+import { modelUserInfoFactory } from "testing/factories/juju/ModelManagerV9";
+import {
+  modelDataInfoFactory,
+  jujuStateFactory,
+  modelDataFactory,
+  modelListInfoFactory,
+} from "testing/factories/juju/juju";
+import { rootStateFactory } from "testing/factories/root";
+import { renderComponent } from "testing/utils";
+
+import ModelActions from "./ModelActions";
+import { Label } from "./types";
+
+describe("ModelActions", () => {
+  it("displays the actions menu", () => {
+    renderComponent(
+      <ModelActions activeUser="eggman@external" modelName="test-model" />,
+    );
+    expect(
+      screen.getByRole("button", { name: "Toggle menu" }),
+    ).toBeInTheDocument();
+  });
+
+  it("shows option to manage access if user has permission", async () => {
+    const state = rootStateFactory.build({
```

Comment:
It looks like the state for this and the test below is the same so it'd probably be good to set it up in a before each.

Reply:
It's generally safer to set up state in a beforeEach so that we get a fresh state object for each test, otherwise anything that mutates state will cause subsequent tests to fail.